File: dev/ClientApp/src/app/models/iranKish/ConfirmInfo.ts

export default class ConfirmInfo {
  constructor(
    public responseCode?: string,
    public status?: boolean,
    public result?: ConfirmResult
  ) { }
}

export class ConfirmResult {
  public responseCode?: string;
  public systemTraceAuditNumber?: number;
  public retrievalReferenceNumber?: number;
  public transactionDateTime?: string;
  public transactionDate?: string;
  public transactionTime?: string;
  public processCode?: string;
  public requestId?: number;
  public additional?: string;
  public billType?: string;
  public billId?: number;
  public amount?: number;
  public revertUri?: string;
  public acceptorId?: number;
  public terminalId?: number;
  public tokenIdentity?: string;
  public isVerified?: boolean;
  public isMultiplex?: boolean;
  public isReversed?: boolean;
  public maskedPan?: string;
  public description?: string;
  public status?: boolean;
  public result?: string;


}

File: Models/DataModels/ConfirmInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace G_IPG_API.Models.DataModels;

public class ConfirmInfo
{
    public ConfirmInfo()
    {
        result = new ConfirmResult();
    }
    public string? responseCode { get; set; }
    public string? description { get; set; }
    public bool? status { get; set; }
    public ConfirmResult? result { get; set; }

}

public class ConfirmResult
{
    public string responseCode { get; set; }
    public int? systemTraceAuditNumber { get; set; }
    public long? retrievalReferenceNumber { get; set; }
    public DateTime transactionDateTime { get; set; }
    public string? transactionDate { get; set; }
    public string? transactionTime { get; set; }
    public string? processCode { get; set; }
    public int? requestId { get; set; }
    public string? additional { get; set; }
    public string? billType { get; set; }
    public long? billId { get; set; }
    public long? amount { get; set; }
    public string revertUri { get; set; }
    public long? acceptorId { get; set; }
    public int? terminalId { get; set; }
    public string tokenIdentity { get; set; }
    public bool? isMultiplex { get; set; }
    public bool? isVerified { get; set; }
    public bool? isReversed { get; set; }
    public string maskedPan { get; set; }
    public string description { get; set; }
    public bool? status { get; set; }
    public string result { get; set; }

}

